airbyte-config/specs/src/main/java/io/airbyte/config/specs/GcsBucketSpecFetcher.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.specs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.api.client.util.Preconditions;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.AirbyteProtocolSchema;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import io.airbyte.validation.json.JsonValidationException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GcsBucketSpecFetcher {
```

Comment:
This class replaces the logic that used to live [inside the GcsBucketSpecCacheSchedulerClient](https://github.com/airbytehq/airbyte/pull/7501/files#diff-75a00421b2ec56faa0c2592689e81f508e3b752dfddf599aec7fac3876a42891L91-L123) because this logic is also needed by the new `SeedConnectorSpecGenerator` script. Moving it was necessary to avoid a circular dependency.